backend/src/main/java/com/mcmasterbaja/analyzer/SGolayFilter.java

```diff
@@ -83,6 +88,8 @@ public void analyze() throws IOException, CsvException {
     ICSVWriter writer = getWriter(outputFiles[0]);
 
     String[] headers = reader.readNext();
+    if (headers==null) { throw new InvalidHeaderException("Failed to read headers from input file: " + inputFiles[0]); }
```

Comment:
I think this is great for now, catching this error is very important. Something to add when we adjust the reader stuff (try with resources https://github.com/McMaster-Baja-Racing/Better-Data-Viewer/issues/151) is to have this built in most likely (i.e. check for empty file)

Reply:
okay!!!!